### PR TITLE
Added prototype air rune, scene change, outofbounds areas, esc button

### PR DESCRIPTION
**Pull request by nikiglen**
This contains everything in the build on Tuesday! Some new prefabs and the addition of a prototype rune to RuneTutorial

### CODE REVIEW

File: Assets/Scripts/AirRune.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        	// Loop through all entities that have entered the effect area
+    		IEnumerator i = inside.GetEnumerator();
+    		while (i.MoveNext())
+    		{
+    			GameObject g = (GameObject)i.Current;
+
+    			// Calculate distance from entity to base
+        		float dist = Vector2.Distance(rune.GetPosition(), g.transform.position);
+        		// Scale force by that distance
+        		float thrust = force;
```
Comment (rej12cool) [inline Assets/Scripts/AirRune.cs]: It doesn't look there's any scaling going on here?

File: Assets/Scripts/GlobalRunes.cs [hunk 2]
```diff
@@ -24,12 +24,17 @@ private struct RuneUpdate
...
     void Start()
     {
-        updateQ = new Queue<RuneUpdate>();
+       
     }
 
     // Update is called once per frame
     void Update()
     {
+        if (updateQ == null)
```
Comment (rej12cool) [inline Assets/Scripts/GlobalRunes.cs]: Why did you move this to the update function? I thought it would have made more sense to initialize it in `Start()` or `Awake()` rather than checking every frame that it's null

File: Assets/Scripts/GlobalRunes.cs [hunk 3]
```diff
@@ -46,16 +51,19 @@ void Update()
...
         Debug.Log("Test successful!");
     }
 
 
     // AddQueue adds a rune and the desired update to the queue
-    public static void AddQueue(GameObject rune, string mode, float rot, Vector2 point)
+    public void AddQueue(GameObject rune, string mode, float rot, Vector2 point)
     {
-        Debug.Log(mode);
+        if (updateQ == null)
```
Comment (rej12cool) [inline Assets/Scripts/GlobalRunes.cs]: Why are you checking again if it's null?

File: Assets/Scripts/Rune.cs [hunk 4]
```diff
@@ -4,6 +4,9 @@
 
 public class Rune : MonoBehaviour
 {
+    // Reference to the script with a queue of all runes
+    public GlobalRunes all_runes;
```
Comment (rej12cool) [inline Assets/Scripts/Rune.cs]: This variable should be named `allRunes`. I think for C# conventions, the variables names generally don't have underscores

File: ProjectSettings/ProjectSettings.asset [hunk 5]
```diff
@@ -13,7 +13,7 @@ PlayerSettings:
   useOnDemandResources: 0
   accelerometerFrequency: 60
   companyName: DefaultCompany
-  productName: Runeshaper
+  productName: Runeshaper 2D
```
Comment (rej12cool) [inline ProjectSettings/ProjectSettings.asset]: Do you know if making this change would cause conflicts with people's local repositories?

**Review by rej12cool — CHANGES_REQUESTED** [review]
See comments

File: ProjectSettings/ProjectSettings.asset (hunk 5 above)
Comment (nikiglen) [reply ProjectSettings/ProjectSettings.asset]: Unsure, changing it to "Runeshaper" just in case

File: Assets/Scripts/Rune.cs (hunk 4 above)
Comment (nikiglen) [reply Assets/Scripts/Rune.cs]: Got it, changed

File: Assets/Scripts/GlobalRunes.cs (hunk 2 above)
Comment (nikiglen) [reply Assets/Scripts/GlobalRunes.cs]: So for this one, I had a lot of issues with an "Object null reference" which I think was occurring because a Rune script was trying to access updateQ before the Start function of GlobalRunes was called. So to fix this, I made it check if it needed to create the queue before it needed to do something with it. There might be a way to make it work for Start, but this solution was simple and fixed the issue.

File: Assets/Scripts/GlobalRunes.cs (hunk 3 above)
Comment (nikiglen) [reply Assets/Scripts/GlobalRunes.cs]: (Same answer as above) So for this one, I had a lot of issues with an "Object null reference" which I think was occurring because a Rune script was trying to access updateQ before the Start function of GlobalRunes was called. So to fix this, I made it check if it needed to create the queue before it needed to do something with it. There might be a way to make it work for Start, but this solution was simple and fixed the issue.

File: Assets/Scripts/AirRune.cs (hunk 1 above)
Comment (nikiglen) [reply Assets/Scripts/AirRune.cs]: Oh that comment is outdated, I was going to use the distance but then opted for just applying the same amount of force every frame. Will eventually use that code, but I just removed it for now. Only has line 35 now

**Review by rej12cool — APPROVED** [review]
